snakeoil/test/test_osutils.py

```diff
@@ -311,44 +312,127 @@ def test_nonexistent(self):
         self.assertRaises(osutils.NonExistent, osutils.FsLock,
                           pjoin(self.dir, 'missing'))
 
-    def test_locking(self):
-        path = pjoin(self.dir, 'lockfile')
+    @staticmethod
```

Comment:
My suggestion above was something of a drive by.  Specifically, for code that can hang indefinitely- wrapping that in an enforced time limit is worth it.  My suggestion was meant as a way to accomplish that, and simplify it- basically pushing the entire locking operation down into a subprocess (which forks as needed to do it's test), containing the test in a fashion that it can time'd out if it hangs.

Something akin to http://include.aorcsik.com/2014/05/28/timeout-decorator/ might also do the trick, depending on if python doesn't temporarily delay SIGALARM w/in the blocking locking code.

Either way, yep, familiar w/ the 'joys' of lockf/flock semantics (and the variances across OS)- my concern is more for the form of the test code.